pkg/completion/completion.go

```diff
@@ -103,39 +134,48 @@ type Completion struct {
 	completed chan struct{}
 
 	// callback is called when Complete is called the first time.
-	callback func()
-}
+	callback func(err error)
 
-// Context returns the context of the asynchronous computation.
-// If the context is cancelled, e.g. if it times out, the computation must be
-// cancelled.
-func (c *Completion) Context() context.Context {
-	return c.ctx
+	// err is the error the completion completed with
+	// Can be accessed without locking only after Complete() returns
```

Comment:
Remove this sentence. This is not true. This field should never be accessed without locking.

Reply:
removed.